src/gsy_e/models/strategy/energy_parameters/energy_params_eb.py

```diff
@@ -21,37 +21,190 @@
+            slots = self.day_forward_slots(market_slot)
+            for slot in slots:
+                self._posted_energy_kWh[slot] -= posted_energy_kWh
+
+    @abstractmethod
+    def serialize(self):
+        """Return dict with the current energy parameter values."""
+
+    @abstractmethod
+    def get_available_energy_kWh(
+            self, market_slot: pendulum.DateTime, market_type: AvailableMarketTypes):
+        """Get the available offer energy of the PV."""
+
+    @abstractmethod
+    def event_activate_energy(self, area):
+        """Initialize values that are required to compute the energy values of the asset."""
+
+    @abstractmethod
+    def event_traded_energy(
+            self, energy_kWh: float, market_slot: pendulum.DateTime,
+            product_type: AvailableMarketTypes):
+        """
+        When a trade happens, we want to split the energy through the entire year following a
+        standard solar profile.
+
+        Args:
+            energy_kWh: the traded energy in kWh.
+            market_slot: the slot targeted by the trade. For forward exchanges, this represents
+                the entire period of time over which the trade profile will be generated.
+            product_type: One of the available market types.
+        """
+
+    @property
+    @abstractmethod
+    def state(self) -> "StateInterface":
+        """Retrieve the internal state of the asset."""
+
+    @staticmethod
+    def day_forward_slots(market_slot: pendulum.DateTime):
+        """Get the market slots for the day forward market."""
+        return [
+            market_slot,
+            market_slot + pendulum.duration(minutes=15),
+            market_slot + pendulum.duration(minutes=30),
+            market_slot + pendulum.duration(minutes=45)]
+
+
+class ConsumptionStandardProfileEnergyParameters(ForwardEnergyParams):
     """Energy parameters of the load strategy for the forward markets."""
 
     def __init__(self, capacity_kW):
+        super().__init__()
         self.capacity_kW: float = capacity_kW
 
         self._state = LoadState()
         self._area = None
         self._profile_generator: Optional[ForwardTradeProfileGenerator] = None
 
+    @property
+    def state(self) -> LoadState:
+        """Retrieve the internal state of the asset."""
+        return self._state
+
     def serialize(self):
         """Return dict with the current energy parameter values."""
         return {
             "capacity_kW": self.capacity_kW
         }
 
+    @property
+    def capacity_kWh(self):
+        """Capacity of the load in kWh."""
+        return convert_kW_to_kWh(self.capacity_kW, self._area.config.slot_length)
+
+    def get_available_energy_kWh(
+            self, market_slot: pendulum.DateTime, market_type: AvailableMarketTypes):
+        """Get the available bid energy of the load."""
+        if market_type == AvailableMarketTypes.INTRADAY:
+            return self._state.get_energy_requirement_Wh(market_slot) / 1000.0
```

Comment:
I am actually not sure whether in these cases it is better to access the private or the public property because there are (dis)advantages to both. Since you prefer the public property though I will replace it.

Reply:
I'm not sure why the public `state` property was created (rather than converting the `_state` variable into a public one). Maybe it was to avoid setting `state` directly, since the property will raise an `AttributeError` in that case?
In any case, if setters/getters are defined for a variable, that's usually done because setting/getting that variable should automatically trigger some other action. For example:
```py
@property
def state(self):
    self.do_something_else()
    return self._state
```
In our case, this doesn't happen. But since we define the property I think it would be better to use that.
What are the disadvantages you see in using the public variable?